docs/developer-docs/smart-contracts/advanced-features/randomness.mdx

```diff
@@ -0,0 +1,85 @@
+# On-Chain Randomness
+
+The Internet Computer provides a secure and verifiable way to generate random numbers directly within canister smart contracts. This functionality is exposed through the [raw_rand](../../../references/ic-interface-spec.md) method offered by the [management canister](../advanced-features/management-canister.mdx) and the [Motoko Random module](https://internetcomputer.org/docs/current/motoko/main/base/Random). 
+
+The method takes no input and returns 32 pseudo-random bytes to the caller.
+
+## How it works
+Behind the scenes, ICP utilizes a Verifiable Random Function (VRF) to generate randomness.
+
+During each round, the VRF is evaluated with the current round number as input. This process produces a fresh set of random bytes.
```

Comment:
Is there a resource that defines what a round number is in case this is a new term to the reader?

Reply:
Hmm, during each round, the VRF is evaluated with the input being the number of the round. Maybe I should use this explanation instead of "round number"?

I also think that linking to the execution layer doc might be interesting to more technical folks: https://internetcomputer.org/how-it-works/execution-layer/